### ISSUE TITLE
🐛 [Bug]  (Found IValue containing object of type int @ at::sum) Encountered bug when using TRTorch

### ISSUE THREAD

**Issue by zhanjw** [NVIDIA/TRTorch#185]
##  Bug Description

Pytorch
```
        x_float = x.float()
        x_mul = x_float.mul(x_float)
        print("x_mul.shape",x_mul.shape)
        norm = x_mul.sum(1, keepdim=True)
        norm = norm.sqrt()
        norm = norm.add(self.eps)
```
Output:
x_mul.shape torch.Size([1, 512, 160, 100])

Compiling:
……
DEBUG: [TRTorch - Debug Build] - Output tensor shape: [1, 256, 1, 1]
DEBUG: [TRTorch Conversion Context] - Evaluating %x_mul.1 : Tensor = aten::mul(%133, %133) # /run/media/eric/DATA/industrial/mmdetection/mmdet/models/backbones/ssd_vgg.py:173:0
DEBUG: [TRTorch Conversion Context] - Found the value to be: 36
INFO: [TRTorch Conversion Context] - Adding Layer %norm.1 : Tensor = aten::sum(%x_mul.1, %135, %7, %134) # /run/media/eric/DATA/industrial/mmdetection/mmdet/models/backbones/ssd_vgg.py:175:0 (ctx.AddLayer)
DEBUG: [TRTorch Conversion Context] - Node input is a result of a previously evaluated value
DEBUG: [TRTorch Conversion Context] - Node input is a result of a previously evaluated value
DEBUG: [TRTorch Conversion Context] - Node input is a result of a previously evaluated value
DEBUG: [TRTorch Conversion Context] - Node input is a result of a previously evaluated value
DEBUG: [TRTorch Conversion Context] - Found IValue containing object of type int
terminate called after throwing an instance of 'trtorch::Error'
  what():  [enforce fail at core/conversion/var/Var.cpp:94] Expected isITensor() || (isIValue() && ptr_.ivalue->isTensor()) to be true but got false
Requested either IValue containing a Tensor, or ITensor, however Var type is c10::IValue

**Found IValue containing object of type int?
But my output[0] is a torch::tensor, how can I solve this problem?**

## To Reproduce

Relevant code
```
        self.l2_norm = L2Norm(
            self.features[out_feature_indices[0] - 1].out_channels,
            l2_norm_scale)
```
```
outs[0] = self.l2_norm(outs[0])
```
```
class L2Norm(nn.Module):

    def __init__(self, n_dims, scale=20., eps=1e-10):
        """L2 normalization layer.

        Args:
            n_dims (int): Number of dimensions to be normalized
            scale (float, optional): Defaults to 20..
            eps (float, optional): Used to avoid division by zero.
                Defaults to 1e-10.
        """
        super(L2Norm, self).__init__()
        self.n_dims = n_dims
        self.weight = nn.Parameter(torch.Tensor(self.n_dims))
        self.eps = eps
        self.scale = scale

    def forward(self, x):
        """Forward function."""
        # normalization layer convert to FP32 in FP16 training
        x_float = x.float()
        norm = torch.sum(x_float*x_float, 1, keepdim=True).sqrt() + self.eps
        # norm = x_float.pow(2).sum(1, keepdim=True).sqrt() + self.eps

        # x_float = x.float()
        # x_mul = x_float.mul(x_float)
        # print("x_mul.shape",x_mul.shape)
        # norm = x_mul.sum(1, keepdim=True)
        # norm = norm.sqrt()
        # norm = norm.add(self.eps)

        return (self.weight[None, :, None, None].float().expand_as(x_float) *
                x_float / norm).type_as(x)
```

## Expected behavior

Complete the compilation

## Environment

> Build information about the TRTorch compiler can be found by turning on debug messages

 - PyTorch Version (e.g., 1.0): 1.5.1
 - CPU Architecture: x86
 - OS (e.g., Linux): Linux(arch)
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source): conda
 - Build command you used (if compiling from source): bazel build //:libtrtorch --compilation_mode dbg --config=pre_cxx11_abi --distdir third_party/dist_dir/x86_64-linux-gnu --verbose_failures
 - Are you using local sources or building from archives: building from archives
 - Python version: 3.7.3
 - CUDA version: 10.2
 - GPU models and configuration: 1050Ti
 - Any other relevant information:

**Comment by peri044** [comment]
I'm unable to reproduce this issue. Infact, I see a different error. The graph that I used to generate the jit file is 
```
x_float = x.float()
x_mul = x_float.mul(x_float)
norm = x_mul.sum(1, keepdim=True)
norm = norm.sqrt()
norm = norm.add(self.eps)
```
The graph looks like 
```
graph(%self.1 : __torch__.L2Norm,
      %x.1 : Tensor):
  %12 : None = prim::Constant() # :0:0
  %11 : bool = prim::Constant[value=1]() # bug.py:38:36
  %3 : bool = prim::Constant[value=0]() # :0:0
  %9 : int = prim::Constant[value=1]() # bug.py:38:25
  %x_float.1 : Tensor = aten::_cast_Float(%x.1, %3) # bug.py:36:18
  %x_mul.1 : Tensor = aten::mul(%x_float.1, %x_float.1) # bug.py:37:16
  %10 : int[] = prim::ListConstruct(%9)
  %norm.1 : Tensor = aten::sum(%x_mul.1, %10, %11, %12) # bug.py:38:15
  %norm0.1 : Tensor = aten::sqrt(%norm.1) # bug.py:39:15
  %18 : float = prim::GetAttr[name="eps"](%self.1)
  %19 : Tensor = aten::add(%norm0.1, %18, %9) # bug.py:40:15
  return (%19)
```
The error that I see is due to a lack of converter.
```
ERROR: [TRTorch - Debug Build] - Method requested cannot be compiled by TRTorch.
Unsupported operators listed below:
  -  aten::add.Scalar(Tensor self, Scalar other, Scalar alpha=1) -> (Tensor)
```

1. Did you implement this converter yourself ? 
2. Can you share your jit graph ? You can paste the output of `torch.jit.load('JIT_FILE').graph` to reproduce the error.

**Comment by zhanjw** [comment]
norm = torch.sum(x_float*x_float, 1, keepdim=True).sqrt() + self.eps
        # norm = x_float.pow(2).sum(1, keepdim=True).sqrt() + self.eps
**->**
`(Found IValue containing object of type int @ at::sum)`

---
        x_float = x.float()
        x_mul = x_float.mul(x_float)
        norm = x_mul.sum(1, keepdim=True)
        norm = norm.sqrt()
        norm = norm.add(self.eps)
->
`Unsupported operators listed below:`
`  -  aten::add.Scalar(Tensor self, Scalar other, Scalar alpha=1) -> (Tensor)`

---
model torchscript file:
https://drive.google.com/file/d/1jYRgr7ALEfCjhJbpW-d0Fsmu7eB82osJ/view?usp=sharing

**Comment by peri044** [comment]
In torch 1.6 and 1.7, I see with both graph variants you mentioned above
```
Unsupported operators listed below:
- aten::add.Scalar(Tensor self, Scalar other, Scalar alpha=1) -> (Tensor)
```

I've raised a PR which fixes this issue [https://github.com/NVIDIA/TRTorch/pull/221](url). I've verified both variants of your graph in Torch 1.6 and 1.7 and they work fine now. 
We have moved from torch 1.5. But I think the JIT graph with `aten::add.Scalar`  might work with torch 1.5 (not verified). Let me know if this resolves your issue.
